src/Sentry/Protocol/SampleProfile.cs

```diff
@@ -0,0 +1,54 @@
+using Sentry.Extensibility;
+using Sentry.Internal;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol;
+
+// A list of frame indexes.
+using SentryProfileStackTrace = HashableGrowableArray<int>;
+
+/// <summary>
+/// Sentry sampling profiler output profile
+/// </summary>
+internal sealed class SampleProfile : IJsonSerializable
+{
+    public GrowableArray<Sample> Samples = new(10000);
+    public GrowableArray<SentryStackFrame> Frames = new(100);
+    public GrowableArray<SentryProfileStackTrace> Stacks = new(100);
+    public List<SentryThread> Threads = new(10);
```

Comment:
Is there any actual benefit in changing to properties, with this being an `internal` class? I thought it's only useful if you need to change the implementation in the future without affecting the interface. I'm fine with it even if it's "only" a convention in this case, just wanted to understand if I'm missing something useful that it would bring.

Reply:
In this particular case, not really - because the class is internal, so the fields are the same as internal even though they're public.  The only risk is that if the class ever gets made public, the fields would become part of the public API.  You could mark them internal now to avoid that.

We generally don't want fields exposed on a public API because it restricts how they can evolve over time.  Rule of thumb is private fields, public properties.  Internals are sorta meh. 😄

Reply:
changed to properties